src/classes/app.py

```diff
@@ -267,24 +273,33 @@ def __init__(self, *args, mode=None):
             # Recover backup file (this can't happen until after the Main Window has completely loaded)
             self.window.RecoverBackup.emit()
 
+    def settings_load_error(self, filepath=None):
+        """Use QMessageBox to warn the user of a settings load issue"""
+        _ = self._tr
+        QMessageBox.warning(
+            None,
+            _("Settings Error"),
+            _("Error loading settings file: %(file_path)s. Settings will be reset.")
+            % {"file_path": filepath}
+            )
+
     def _tr(self, message):
         return self.translate("", message)
 
     # Start event loop
     def run(self):
         """ Start the primary Qt event loop for the interface """
+        return self.exec_()
 
-        res = self.exec_()
-
+    @pyqtSlot()
+    def cleanup(self):
+        """aboutToQuit signal handler for application exit"""
         try:
             from classes.logger import log
             self.settings.save()
         except Exception:
             log.error("Couldn't save user settings on exit.", exc_info=1)
 
-        # return exit result
-        return res
```

Comment:
@jonoomph 

Oh, I had mean to explain this change some as well.

Qt [advise against](https://doc.qt.io/qt-5/qapplication.html#exec) relying on `exec()` to return on program exit and trying to run any code after that point, because there are scenarios where it doesn't happen:

> We recommend that you connect clean-up code to the `aboutToQuit()` signal, instead of putting it in your application's main() function. This is because, on some platforms the `QApplication::exec()` call may not return. For example, on the Windows platform, when the user logs off, the system terminates the process after Qt closes all top-level windows. Hence, there is no guarantee that the application will have time to exit its event loop and execute code at the end of the `main()` function, after the `QApplication::exec()` call.


So, now we follow that advice.